src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java

```diff
@@ -39,34 +42,53 @@ public class AddAppointmentCommand extends Command {
     private final String description;
     private final LocalDateTime dateTime;
 
+    private final Index doctorIndex;
+
     /**
      * Creates an AddCommand to add the specified {@code Person}
      */
-    public AddAppointmentCommand(Index targetIndex, String description, LocalDateTime dateTime) {
+    public AddAppointmentCommand(Index targetIndex, Index doctorIndex, String description, LocalDateTime dateTime) {
         this.targetIndex = targetIndex;
         this.description = description;
         this.dateTime = dateTime;
+        this.doctorIndex = doctorIndex;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
+        List<Doctor> lastDoctorList = model.getFilteredDoctorList();
+
+        if (lastDoctorList.size() == 0) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
 
         if (targetIndex.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
-        Person targetPatient = lastShownList.get(targetIndex.getZeroBased());
+        if (doctorIndex.getZeroBased() > lastDoctorList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
 
-        Appointment toAdd = new Appointment(description, dateTime, targetPatient);
+        Person targetPatient = lastShownList.get(targetIndex.getZeroBased());
+        Doctor targetDoctor = lastDoctorList.get(doctorIndex.getZeroBased());
+        String name = targetDoctor.getName().toString();
+        Appointment toAdd = new Appointment(description, dateTime, targetPatient, name);
 
         if (targetPatient.hasAppointment(toAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_APPOINTMENT);
         }
 
+        if (targetDoctor.hasAppointment(toAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_APPOINTMENT);
```

Comment:
Create new duplicate message for duplicate doctors
Current message is "This appointment already exists for the patient."
Should be "This doctor already has an appointment at the same time."